### ISSUE TITLE
NullReferenceException when assigning result of awaited promise to variable

### ISSUE THREAD

**Issue by rmja** [nilproject/NiL.JS#238]
I think this is related to #213.

The following code throws:

```
var script = Script.Parse(@"
export default class Test {
    async run() {
        const someVariable = await Promise.resolve(123);
    }
}
");
                var context = new GlobalContext();
                var module = new Module($"main.js", script, context);
                module.Run();

                var ctor = module.Exports.Default.As<Function>();
                var instance = ctor.Construct(new Arguments());
                var run = instance.GetProperty("run").As<Function>().MakeDelegate<Func<JSValue>>();

                var result = await run().As<Promise>().Task;
```

It seems that [this line](https://github.com/nilproject/NiL.JS/blob/develop/NiL.JS/Expressions/Assignment.cs#L32) has `temp` being assigned to null:
![image](https://user-images.githubusercontent.com/2112306/115504518-62a9c300-a278-11eb-828e-3d9d89332263.png)

Which then later fails when `field` is assigned with `temp`.

**Comment by rmja** [comment]
It is not only for variable assignment. This throws the same exception:
```
export default class Test {
    async run() {
        // Note that the awaited promise is added to an array before it is returned.
        return [await Promise.resolve(123)];
    }
}
```

**Comment by nilproject** [comment]
Thanks for this bug report!

As temporal workaround you can bring class out of `export` statement

**Comment by rmja** [comment]
@nilproject Would it be possible to get a release with this fix? - together with #239 being fixed it seems that I have a fully working setup:)

**Comment by nilproject** [comment]
@rmja sorry for my so long lag. I will upload the new version now
